lib/purl_finder.rb

```diff
@@ -85,7 +85,11 @@ def index_purls(params={})
       druid = get_druid_from_file_path(line)
       if !druid.blank?
         IndexingLogger.info("indexing #{druid}")
-        result = Purl.save_from_public_xml(File.dirname(line)) # pass the directory of the file containing public
+        begin
+          result = Purl.save_from_public_xml(File.dirname(line)) # pass the directory of the file containing public
+        rescue => e
+          IndexingLogger.error("An error occurred while trying to save #{druid}.  \n#{e.message} \n#{e.backtrace.inspect}")
+        end
```

Comment:
Or maybe it's better to change the `Purl.save_from_public_xml` to thrown an exception (which may just be changing `.save` to `.save!`?) rather than return `true` or `false`? Then we could write something like this:

```ruby
begin
  Purl.save_from_public_xml(...)
  success += 1
rescue => e
  IndexingLogger.error(...)
  error += 1
end
```